packages/fast-usdc/src/exos/advancer.js

```diff
@@ -9,116 +11,219 @@ import { addressTools } from '../utils/address.js';
-      handleTransactionEvent(evidence) {
-        // TODO EventFeed will perform input validation checks.
-        const { recipientAddress } = evidence.aux;
-        const { EUD } = addressTools.getQueryParams(recipientAddress).params;
-        if (!EUD) {
-          statusManager.observe(evidence);
-          throw makeError(
-            `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
-          );
-        }
+      advancer: {
+        /**
+         * Returns a Promise for a Vow in the happy path, or undefined
+         * when conditions are not met. Aims to perform a status update for
+         * every observed transaction.
+         *
+         * We do not expect any callers to depend on the settlement of
+         * `handleTransactionEvent` - errors caught are communicated to the
+         * `StatusManager` - so we don't need to concern ourselves with
+         * preserving the vow chain for callers.
+         *
+         * @param {CctpTxEvidence} evidence
+         */
+        async handleTransactionEvent(evidence) {
+          await null;
+          try {
+            // TODO poolAccount might be a vow we need to unwrap
+            const { assetManagerFacet, poolAccount, usdcBrand } = this.state;
+            const { recipientAddress } = evidence.aux;
+            const { EUD } =
+              addressTools.getQueryParams(recipientAddress).params;
+            if (!EUD) {
+              throw makeError(
+                `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
+              );
+            }
 
-        // TODO #10391 this can throw, and should make a status update in the catch
-        const destination = chainHub.makeChainAddress(EUD);
+            // this will throw if the bech32 prefix is not found, but is handled by the catch
+            const destination = chainHub.makeChainAddress(EUD);
 
-        /** @type {DenomAmount} */
-        const requestedAmount = harden({
-          denom: this.state.localDenom,
-          value: BigInt(evidence.tx.amount),
-        });
+            const requestedValue = BigInt(evidence.tx.amount);
+            const requestedAmount = AmountMath.make(usdcBrand, requestedValue);
+            const poolBalance = assetManagerFacet.lookupBalance();
 
-        // TODO #10391 ensure there's enough funds in poolAccount
+            if (!AmountMath.isGTE(poolBalance, requestedAmount)) {
+              log(
+                `Insufficient pool funds`,
+                `Requested ${q(requestedAmount)} but only have ${q(poolBalance)}`,
+              );
+              statusManager.observe(evidence);
+              return;
+            }
 
-        const transferV = E(this.state.poolAccount).transfer(
-          destination,
-          requestedAmount,
-        );
+            try {
+              // mark as Advanced since `transferV` initiates the advance
+              // will throw if we've already .skipped or .advanced this evidence
```

Comment:
without punctuation or capitalization this reads like a single sentence

Reply:
thanks, addressed